compose-builder/Makefile

```
@@ -1034,8 +1052,11 @@ pull: gen
 
 gen:
 	echo MQTT_VERBOSE=${MQTT_VERBOSE}
-	${DOCKER_COMPOSE}  -p edgex $(COMPOSE_FILES) ${GEN_COMMAND} > docker-compose.yml
-	rm -rf ./$(GEN_EXT_DIR)
+	${DOCKER_COMPOSE}  -p edgex $(COMPOSE_FILES) ${GEN_COMMAND} > docker-compose.yml.gen
+	if [ "$(NO_CLEANUP)" = "0" ]; then rm -rf ./$(GEN_EXT_DIR); fi
+	echo "# Generated with: $(docker_ver)" > docker-compose.yml
```

Doesn't this overwrite the file with just the version? Probably need to be moved above the gen and switch the gen to be a >>
```suggestion
	echo "# Generated with: $(docker_ver)" > docker-compose.yml
```

---

yes but then it gets appended to below from the generated file, and then that generated file is removed, resulting in a file with a 'header' of what docker version generated it and the rest of the contents in this file:
```
	echo "# Generated with: $(docker_ver)" > docker-compose.yml
	cat docker-compose.yml.gen >> docker-compose.yml
	rm docker-compose.yml.gen
```

the second line cats the generated file into the resultant

---

oh i see what you're thinknig there... sure i can adapt